keel-artifact/src/main/kotlin/com/netflix/spinnaker/keel/artifacts/ArtifactListener.kt

```diff
@@ -53,15 +54,15 @@ class ArtifactListener(
       .forEach { artifact ->
         if (repository.isRegistered(artifact.name, artifact.artifactType)) {
           val artifactSupplier = artifactSuppliers.supporting(artifact.artifactType)
-          val version = artifactSupplier.getFullVersionString(artifact)
-          var status = artifactSupplier.getReleaseStatus(artifact)
-          log.info("Registering version {} ({}) of {} {}", version, status, artifact.name, artifact.type)
-          repository.storeArtifact(artifact.name, artifact.artifactType, version, status)
+          val enrichedArtifact = artifactSupplier.addMetadata(artifact.normalized())
+
+          log.info("Registering version {} (status={}) of {} artifact {}",
+            artifact.version, artifact.status, artifact.type, artifact.name)
+
+          repository.storeArtifactVersion(enrichedArtifact)
```

Comment:
Instead of doing one write to `ARTIFACT_VERSIONS` to store the basics here, and another as a reaction to the `ArtifactSaved` event below, consolidate in a single call to `storeArtifactVersion()`.

Reply:
What if the call to get metadata fails? Then we won't write the version of the artifact at all. It seems more reliable to first write the artifact, then add the metadata to it if it exists

Reply:
Ah, I see, you've wrapped the get metadata in exception handling and we will save the original artifact. That seems fine.

Reply:
Yep. There should be no change in behavior there. That's what we were doing in the other listener already.